src/models/hooks/goalTemplate.js

```diff
@@ -0,0 +1,74 @@
+// import { auditLogger } from '../../logger';
+import { CREATION_METHOD } from '../../constants';
+
+const { Op } = require('sequelize');
+
+const autoPopulateHash = (sequelize, instance) => {
+  const changed = instance.changed();
+  if (Array.isArray(changed)
+    && changed.includes('templateName')
+    && instance.templateName !== null
+    && instance.templateName !== undefined) {
+    instance.set('hash', sequelize.fn('md5', sequelize.fn('TRIM', instance.templateName)));
```

Comment:
Should we also lowercase/uppercase the `templateName` so `GOAL` AND `goal` have the same hash?

Reply:
We are going to refine the comparison more in the next phase, but lower could prevent some corrections from being saved.